openwisp_utils/utils.py

```diff
@@ -67,14 +67,39 @@ def print_color(string, color_name, end='\n'):
     print(f'\033[{color}m{string}\033[0m', end=end)
 
 
-def retryable_request(method, timeout=(4, 8), max_retries=3, **kwargs):
-    request_session = requests.Session()
-    retries = Retry(
-        total=max_retries,
-        backoff_factor=1,
-        status_forcelist=[429, 500, 502, 503, 504],
-        allowed_methods=['DELETE', 'GET', 'HEAD', 'OPTIONS', 'PUT', 'TRACE', 'POST'],
+def retryable_request(
+    method,
+    timeout=(4, 8),
+    max_retries=3,
+    backoff_factor=1,
+    backoff_jitter=0.0,
+    status_forcelist=(429, 500, 502, 503, 504),
+    allowed_methods=('HEAD', 'GET', 'PUT', 'DELETE', 'OPTIONS', 'TRACE', 'POST'),
```

Comment:
one more time, **let's not use mutable default arguments** for the reasons described here:
https://dev.to/sharmapacific/mutable-default-arguments-in-python-22ce

We have already been hit a few times by obscure bugs which originated from using mutable default arguments, let's not repeat the same mistakes.

This applies to timeout, status_forcelist, allowed_methods.